lib/Parser/Parser.cpp

```diff
@@ -203,6 +205,30 @@ Token Lexer::getNextToken(std::string &ErrStr) {
                         StringRef(NumBegin, NumEnd - NumBegin), 10)};
   }
 
+  if (*Begin == '(') {
+    ++Begin;
+    const char *NumBegin = Begin;
+    while (Begin != End && (*Begin == '0' || *Begin == '1' || *Begin == 'x')) {
+      ++Begin;
+    }
+    const char *NumEnd = Begin - 1;
+    if (Begin == NumBegin) {
+      ErrStr = "invalid knownbits string";
+      return Token{Token::Error, Begin, 0, APInt()};
+    }
+    Token T;
+    T.K = Token::KnownBits;
+    T.Pos = NumBegin;
+    T.Len = size_t(Begin - NumBegin);
+    T.PatternString = StringRef(NumBegin, NumEnd - NumBegin + 1); 
+    if (Begin != End && *Begin != ')') {
```

Comment:
Begin != End here is necessary just to make sure that there is no new line character in between. For example, if someone writes : %0:i4 = var (10\n01)\n
Everytime new line character comes, it goes to new line and the scanning of tokens would be different and we will no longer have a track of if its going to be a part of known bits string or not. This is how in the exisiting parser, we always look for '\n' always.